wctl/tx.go

```diff
@@ -0,0 +1,263 @@
+package wctl
+
+import (
+	"encoding/hex"
+	"errors"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/perlin-network/noise/edwards25519"
+	"github.com/valyala/fastjson"
+)
+
+var (
+	_ UnmarshalableJSON = (*TxResponse)(nil)
+	_ UnmarshalableJSON = (*Transaction)(nil)
+	_ UnmarshalableJSON = (*TransactionList)(nil)
+	_ MarshalableJSON   = (*TxRequest)(nil)
+)
+
+var (
+	// ErrInsufficientPerls is returned when you don't have enough PERLs.
+	ErrInsufficientPerls = errors.New("Insufficient PERLs")
+)
+
+type TransactionEvent struct {
+	Event   string    `json:"event"`
+	ID      [32]byte  `json:"tx_id"`
+	Sender  [32]byte  `json:"sender_id"`
+	Creator [32]byte  `json:"creator_id"`
+	Depth   uint64    `json:"depth"`
+	Tag     byte      `json:"tag"`
+	Time    time.Time `json:"time"`
+}
+
+type Transaction struct {
+	ID      [32]byte `json:"id"`
```

Comment:
Would be ideal instead to make `wavelet.Transaction` JSON marshable in this case. What do you think?

Reply:
While the Transaction in Wavelet is used for internal workings, the Transaction in wctl is exposed as the API response, so I don't think this is a good idea.